### PR TITLE
offset_get ignores --storage option when checking group existence

### PR DESCRIPTION
**Pull request by neoeahit**
offset_get ignores --storage option when checking group existence

This will check if storage option is kafka, and in case it is, it will read list of topics form kafka and not from zookeeper. Default option is still zookeeper.

### CODE REVIEW

File: kafka_utils/kafka_consumer_manager/util.py [hunk 1]
```diff
@@ -113,3 +126,91 @@ def prompt_user_input(in_str):
             sys.exit(0)
         if answer == "y" or answer == "yes":
             return
+
+
+class InvalidMessageException(Exception):
+    pass
+
+
+class KafkaGroupReader:
```
Comment (neoeahit) [inline kafka_utils/kafka_consumer_manager/util.py]: This method was untested, in moving it around, I hv

File: kafka_utils/kafka_consumer_manager/commands/offset_manager.py [hunk 2]
```diff
@@ -77,11 +79,15 @@ def preprocess_args(
                     groupid=groupid,
                 ),
             )
-        topics = cls.get_topics_from_consumer_group_id(
-            cluster_config,
-            groupid,
-            fail_on_error,
-        )
+        if storage == 'zookeeper':
```
Comment (ecanzonieri) [inline kafka_utils/kafka_consumer_manager/commands/offset_manager.py]: Should this condition be inside get_topics_from_consumer_group_id?

File: kafka_utils/kafka_consumer_manager/commands/offset_manager.py [hunk 3]
```diff
@@ -77,11 +79,15 @@ def preprocess_args(
...
+            topics = cls.get_topics_from_consumer_group_id(
+                cluster_config,
+                groupid,
+                fail_on_error,
+            )
+        else:
+            kafka_group_reader = KafkaGroupReader(cluster_config)
```
Comment (ecanzonieri) [inline kafka_utils/kafka_consumer_manager/commands/offset_manager.py]: I think get_topics_from_consumer_group_id should be able to fetch from either Kafka or Zookeeper. We should be able to move this logic inside that function.

File: tests/kafka_consumer_manager/test_utils.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
+import mock
+from kafka.common import ConsumerTimeout
+
+from kafka_utils.kafka_consumer_manager.util import KafkaGroupReader
+from kafka_utils.util.offsets import PartitionOffsets
+
+
+class TestUtils(object):
```
Comment (ecanzonieri) [inline tests/kafka_consumer_manager/test_utils.py]: Rename to TestKafkaGroupReader

File: tests/kafka_consumer_manager/test_utils.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+                    0,
+                    None
+                ]
+        ):
+            kafka_group_reader.process_consumer_offset_message('test message')
+            assert kafka_group_reader.kafka_groups == {}
+
+    def test_read_groups(self):
+
+        class Mock_Consumer(object):
```
Comment (ecanzonieri) [inline tests/kafka_consumer_manager/test_utils.py]: We don't really need this class in this test. We should be able to use mock_consumer.return_value.next.return_value = mock.Mock(partition=0, topic='test_topic')

File: tests/kafka_consumer_manager/test_utils.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+                        45
+                    ]
+                ):
+                    kafka_group_reader.read_groups()
+                    assert kafka_group_reader.kafka_groups['test_group'] == {"test_topic"}
+                    assert len(kafka_group_reader.finished_partitions) == 1
+
+    def test_read_groups_with_consumer_timeout(self):
+
+        class Mock_Consumer(object):
```
Comment (ecanzonieri) [inline tests/kafka_consumer_manager/test_utils.py]: Same here. We should be able to use side_effect to raise an exception

File: setup.py [hunk 7]
```diff
@@ -55,6 +55,7 @@
         "requests-futures>0.9.0",
         "kafka-python<1.0.0",
         "requests<3.0.0",
+        'retrying'
```
Comment (ecanzonieri) [inline setup.py]: Please refresh requirements.txt as well.

**Pull request merged**